File: estudos-pf/nivel-2-sequencias-e-comprehensions/solution/q1/q1.clj
;; list(map(lambda n: n * 10, filter(lambda n: n > 0, numeros)))

(defn comprehension [lambda sequencia filtro]
  (for [n sequencia :when (filtro n)] (lambda n)))

(println (comprehension #(* % 10) '(-2 -1 0 1 2) #(> % 0)))


  

File: estudos-pf/nivel-2-sequencias-e-comprehensions/solution/q1/q1.py
numeros = [-1, -2, 0, 1, 2]

list_comprehension = [n * 10 for n in numeros if n > 0]

lista_map_filter = list(map(lambda n: n * 10, filter(lambda n: n > 0, numeros)))

assert list_comprehension == lista_map_filter
